src/visualizers/VisualizerFactorHistogram.ts

```diff
@@ -0,0 +1,200 @@
+import {VisualizerExportModule} from '@/visualizers/VisualizerInterface'
+import {VisualizerDefault} from './VisualizerDefault'
+import {natlog} from '../shared/math'
+
+/** md
+# Factor Histogram
+
+(example image should go here)
+
+This visualizer takes the prime factorization of each element in the sequence
+and then creates a histogram of the number of elements of the sequence
+with X prime factors. the number of elements with X prime factors is along
+the x-axis. And the height is how many elements are in each bin of X.
+
+## Parameters
+**/
+
+class FactorHistogramVisualizer extends VisualizerDefault {
+    name = 'Histogram'
+
+    binSize = 1
+    terms = 100
+    firstIndex = this.seq.first
+
```

Comment:
At visualizer construction time, there is no seq yet, so you can't get the correct first index this way. You have to have some value for your firstIndex parameter that means "take the sequence's first index" and then in setup if that's the value of the parameter you can grab the first index from the seq, which will be valid by then. Again, check out VisualizerChaos for an example. Thanks.